### PR TITLE
numpy_split support param type fix; numpy_vsplit added

### PR DESCRIPTION
**Pull request by zoeygxy**
## Description ##
Added python list and set as supported types for parameter 'indices_or_sections'
Added python frontend support for [numpy.vsplit](https://docs.scipy.org/doc/numpy/reference/generated/numpy.vsplit.html)

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Added python list and set as supported types for parameter 'indices_or_sections'
- [x] Added python frontend support for [numpy.vsplit](https://docs.scipy.org/doc/numpy/reference/generated/numpy.vsplit.html)

## Comments ##
- Note: numpy.split and numpy.vsplit still does not support type ndarray for parameter 'indices_or_sections'.

### CODE REVIEW

**Comment by zoeygxy** [comment]
@haojin2 Please review the changes. Thanks!

**Comment by zoeygxy** [comment]
Documentation updated.

File: python/mxnet/numpy/multiarray.py [hunk 1]
```diff
@@ -1835,6 +1835,83 @@ def split(ary, indices_or_sections, axis=0):
...
+    split : Split an array into multiple sub-arrays of equal size.
+
+    Notes
+    -------
+    This function differs from the original `numpy.degrees
+    <https://docs.scipy.org/doc/numpy/reference/generated/numpy.degrees.html>`_ in
+    the following aspects:
+
+    - Currently parameter ``indices_or_sections`` does not support ndarray, but supports scalar,
+    tuple
```
Comment (haojin2) [inline python/mxnet/numpy/multiarray.py]: `" and list"` ?

Comment (zoeygxy) [reply python/mxnet/numpy/multiarray.py]: Good catch! Fixed.

**Comment by zoeygxy** [comment]
This PR will be directly merged to master branch. (#15983)

**Pull request closed without merging**